engines/private/private.cpp

```diff
@@ -0,0 +1,1120 @@
+    _playedPhoneClips.clear();
+
+    // Movies
+
+    _repeatedMovieExit = NULL;
+    _playedMovies.clear();
+
+}
+
+Common::Error PrivateEngine::loadGameStream(Common::SeekableReadStream *stream) {
+    Common::Serializer s(stream, nullptr);
+    debug("loadGameStream");
+    _nextSetting = &kStartGame;
+    int val;
+
+    for (NameList::iterator it = variableList.begin(); it != variableList.end(); ++it) {
+        s.syncAsUint32LE(val);
+        Private::Symbol *sym = variables.getVal(*it);
+        sym->u.val = val;
+    }
+
+    // Diary
+
+    for (NameList::iterator it = locationList.begin(); it != locationList.end(); ++it) {
+        s.syncAsUint32LE(val);
+        Private::Symbol *sym = locations.getVal(*it);
+        sym->u.val = val;
+    }
+
+    uint32 size = 0;
+    size = stream->readUint32LE();
+    Common::String *file = NULL;
+    for (uint32 i = 0; i < size; ++i) {
+        file = new Common::String(stream->readString());
+        inventory.push_back(*file);
+    }
+
+    // Dossiers
+    size = stream->readUint32LE();
+    for (uint32 i = 0; i < size; ++i) {
+        file = new Common::String(stream->readString());
+
+        DossierInfo *m = (DossierInfo*) malloc(sizeof(DossierInfo));
+        m->page1 = file;
+
+        file = new Common::String(stream->readString());
+        if (file->size() == 0) {
+            m->page2 = NULL;
+        } else {
+            m->page2 = file;
+        }
+        _dossiers.push_back(*m);
+    }
+
+    // Radios
+
+    Common::String *sound;
+    size = stream->readUint32LE();
+    debug("AMRadio size %d", size);
+    _AMRadio.clear();
+
+    for (uint32 i = 0; i < size; ++i) {
+        sound = new Common::String(stream->readString());
+        debug("sound: %s", sound->c_str());
+        _AMRadio.push_back(*sound);
+    }
+
+    size = stream->readUint32LE();
+    debug("policeRadio size %d", size);
+    _policeRadio.clear();
+
+    for (uint32 i = 0; i < size; ++i) {
+        sound = new Common::String(stream->readString());
+        debug("sound: %s", sound->c_str());
+        _policeRadio.push_back(*sound);
+    }
+
+    size = stream->readUint32LE();
+    _phone.clear();
+
+    for (uint32 j = 0; j < size; ++j) {
+        PhoneInfo *i = (PhoneInfo*) malloc(sizeof(PhoneInfo));
+
+        i->sound = new Common::String(stream->readString());
+        i->flag  = variables.getVal(stream->readString());
+        i->val   = stream->readUint32LE();
+        _phone.push_back(*i);
+    }
+
+    *_repeatedMovieExit = stream->readString();
+
+    _playedMovies.clear();
+    size = stream->readUint32LE();
+    Common::String *movie;
+
+    for (uint32 i = 0; i < size; ++i) {
+        movie = new Common::String(stream->readString());
+        _playedMovies.setVal(*movie, true);
+    }
+
+    _playedPhoneClips.clear();
+    size = stream->readUint32LE();
+    Common::String *phone;
+
+    for (uint32 i = 0; i < size; ++i) {
+        phone = new Common::String(stream->readString());
```

Comment:
I think you're massively leaking memory here.

Reply:
Is this better?

```cpp
    for (uint32 i = 0; i < size; ++i) {
        Common::String phone = stream->readString();
        ...
```

Reply:
You do not need `phone` at all. Just push the result of reading from the file right away.

Reply:
`phone` and the leaking were removed